### PR TITLE
chore: drop explicit `check_identity`

### LINKED ISSUE #1414: refactor: drop `check_identity` from ExecutionSpec
CheckIdentity should be enabled for all specifications, we don't need this configuration in ExecutionSpec.

### PR DESCRIPTION
**Pull request by 35C4n0r**
**Summary:**  
_Briefly describe the changes made in this PR._

**Issue Reference(s):**  
Fixes #1414
/claim #1414 

**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

## Summary by CodeRabbit

- **Tests**
	- Updated test instructions and specifications for more clarity on expected behaviors.
	- Modified test configurations to reflect changes in schema annotations and directive orders.
	- Introduced new expressions for logical operations and reordered existing ones for consistency.
	- Adjusted schema declarations across various tests to include new directives and reorganize type definitions for improved clarity.
	- Reordered properties within objects to match updated schema requirements.
- **Documentation**
	- Clarified the purpose of specific test instructions and the impact of schema changes on server block evaluations.
	- Restructured and optimized schema annotations in GraphQL files for server-side processing.

### CODE REVIEW

File: tests/execution_spec.rs [hunk 1]
```diff
@@ -33,6 +33,7 @@
 use tailcall::{EnvIO, FileIO, HttpIO};
 use telemetry::in_memory::InMemoryTelemetry;
 use telemetry::init::init_opentelemetry;
+// use std::panic::{self, AssertUnwindSafe};
```
[the remark this answers is not in the document]
Comment (35C4n0r) [reply tests/execution_spec.rs:36]: What do you think is Identity check and why are the tests failing?

[the remark this answers is not in the document]
Comment (35C4n0r) [reply tests/execution_spec.rs]: the tests fail, in all of them (graphql and json server) there are either changes in the order of the parameters, methods or type declarations + schema --> schema @server @upstream

[the remark this answers is not in the document]
Comment (35C4n0r) [reply tests/execution_spec.rs]: @coderrabitai the tests fail, in all of them (graphql and json server) there are either changes in the order of the parameters, methods or type declarations + schema --> schema @server @upstream

**Comment by tusharmath** [comment]
@35C4n0r We need to fix the failing unit tests.

**Comment by tusharmath** [comment]
It should be easier to fix this once we have https://github.com/tailcallhq/tailcall/issues/1461

**Comment by 35C4n0r** [comment]
@tusharmath can you remove `ci:lint` for a while

**Comment by 35C4n0r** [comment]
@tusharmath I've fixed the tests for now, but if we lint the md files tests will fail, 
```
assertion failed: `(left == right)`: Identity check failed for "/home/jay/algora/tailcall/tests/execution/add-field-modify.md"

Diff < left / right > :
 schema @server @upstream {
   query: Query
 }
 
 type Address {
   city: String
   street: String
   zipcode: String
 }
 
 type Query {
   user: User @http(baseURL: "http://jsonplaceholder.typicode.com", path: "/users/1")
 }
 
<type User
<  @addField(name: "street", path: ["address", "street"])
<  @addField(name: "city", path: ["address", "city"])
<  @addField(name: "zipcode", path: ["address", "zipcode"]) {
>type User @addField(name: "street", path: ["address", "street"]) @addField(name: "city", path: ["address", "city"]) @addField(name: "zipcode", path: ["address", "zipcode"]) {
   address: Address
   name: String
 }


note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
test test ... FAILED
```

**Comment by tusharmath** [comment]
closed in favour of https://github.com/tailcallhq/tailcall/pull/1424

**Comment by 35C4n0r** [comment]
Too many conflicts, closing this in favor of #1506

**Pull request closed without merging**